swiftemulator/emulators/base.py

```diff
@@ -123,3 +123,90 @@ def predict_values(
             )
 
         raise NotImplementedError
+
+    def interactive_plot(
+        self,
+        x: np.array,
+        xlabel: str = "",
+        ylabel: str = "",
+        x_data: np.array = None,
+        y_data: np.array = None,
+    ):
+        """
+        Generates an interactive plot which displays the emulator predictions.
+        If no reference data is passed to be overplotted then the plot will
+        display a line which corresponds to the predictions for the mean
```

Comment:
Is it the mean or the midpoint of the ranges? (also out of curiosity, I guess it's the same for uniform points)